### DISCUSSION TITLE
Workspace switch gesture with Mod+MMB

### DISCUSSION THREAD

**Discussion by Vermoot** [YaLTeR/niri#387] [Ideas]
Hi!
The very same way you can already move across a strip with horizontal movement with Mod+MMB, I'd like to be able to switch workspaces with vertical movement.

For good UX, I think there should be a threshold (possibly configurable?) so that you'd need to move a certain amount vertically before it happens. That way you can keep moving horizontally without having the workspace switch come into effect.

**AND/OR**, probably a good thing would be to have only vertical movement or horizontal movement happen at once, never at the same time. I'm not exactly sure which solution is better or what the actual feel should be.

**Comment by Vermoot** [comment]
I guess the best solution would be the second one, so when you Mod+MMB, if your first movement is lateral you move horizontally through columns, just like it's already possible to do, but if your first movement is vertical you move vertically to the adjacent workspace.

And if you move vertically, the experience would be similar to the horizontal scrolling: the view follows your mouse smoothly, and when you release the mouse button it centers on the workspace which is most in view

**Reply by YaLTeR** [reply]
Yeah, that's already how it works with the touchpad swipe gesture. Can do the same with mouse.

**Comment by YaLTeR** [comment]
Implemented in db89d4d3dd1440e18193d4ce98d207531c7720e0
